cmd/tailscale/main.go

```diff
@@ -370,16 +375,19 @@ func (a *App) runBackend(ctx context.Context) error {
 		case n := <-notifications:
 			exitWasOnline := state.ExitStatus == ExitOnline
 			if p := n.Prefs; p != nil && n.Prefs.Valid() {
-				state.Prefs = p.AsStruct()
+				// make a copy of prefs and use the readonly view
+				state.Prefs = p.AsStruct().View()
```

Comment:
Why do we need to copy the PrefsView if we're just getting another read-only view to them anyway?